### DISCUSSION TITLE
BUILD_EPUB_PREVIEW

### DISCUSSION THREAD

**Discussion by taunosiitam** [sumatrapdfreader/sumatrapdf#4510] [General]
Hey! I checked out and built the vs2022 solution yesterday to understand what I need to do to get EPUB previews working. It turned out BUILD_EPUB_PREVIEW define, which is used to include the functionality, has been there since at least 2 years ago when Visual Studio 2022 solution was added. So far it seems to be used in debug builds only. I then added that define for a release build and rebuilt. Because I didn't immediately understand how to build an installer, I just copied the new binaries over my existing installation at `C:\Program Files\SumatraPDF` and created a [REG file](https://github.com/user-attachments/files/16862928/sumatrapdf_epub_preview.zip) to update the registry. And I can see EPUB previews now. It has some faults, like scrolling too fast through the pages makes the preview go blank, for example, but I'm already happy with what I'm seeing.

My questions are: why it has not made to a release version yet? I understand there are still bugs present, have those been mapped out and listed anywhere? Are those all [mupdf](https://github.com/ArtifexSoftware/mupdf) bugs and not related to SumatraPDF? What plans do you have with EPUB preview functionality? Thanks!

**Comment by GitHubRulesOK** [comment]
I thought it was in pre-release but on checking it is only the icon not the pre-view. The main Issue has generally been "variability" of results since the "First" page is not always a cover image and if an image at what scale to fit?

**Comment by kjk** [comment]
Mostly being overly cautious. Probably should enable it.
